File: security/src/main/java/top/bettercode/summer/security/ApiToken.java
package top.bettercode.summer.security;

import java.io.Serializable;
import org.springframework.security.core.userdetails.UserDetails;

/**
 * @author Peter Wu
 */
public class ApiToken implements Serializable {

  private static final long serialVersionUID = 1L;

  private String scope;

  private Token accessToken;

  private Token refreshToken;

  private  InstantAt userDetailsInstantAt;

  private UserDetails userDetails;

  public ApiToken() {
  }

  public ApiToken(String scope, Token accessToken,
      Token refreshToken, InstantAt userDetailsInstantAt, UserDetails userDetails) {
    this.scope = scope;
    this.accessToken = accessToken;
    this.refreshToken = refreshToken;
    this.userDetailsInstantAt = userDetailsInstantAt;
    this.userDetails = userDetails;
  }

  public String getUsername() {
    return userDetails.getUsername();
  }

  public ApiAccessToken toApiToken(){
    return new ApiAccessToken(this);
  }

  //--------------------------------------------

  public String getScope() {
    return scope;
  }

  public void setScope(String scope) {
    this.scope = scope;
  }

  public Token getAccessToken() {
    return accessToken;
  }

  public void setAccessToken(Token accessToken) {
    this.accessToken = accessToken;
  }

  public Token getRefreshToken() {
    return refreshToken;
  }

  public void setRefreshToken(Token refreshToken) {
    this.refreshToken = refreshToken;
  }

  public InstantAt getUserDetailsInstantAt() {
    return userDetailsInstantAt;
  }

  public void setUserDetailsInstantAt(
      InstantAt userDetailsInstantAt) {
    this.userDetailsInstantAt = userDetailsInstantAt;
  }

  public UserDetails getUserDetails() {
    return userDetails;
  }

  public void setUserDetails(UserDetails userDetails) {
    this.userDetails = userDetails;
  }

}

File: security/src/main/kotlin/top/bettercode/summer/security/token/ApiToken.kt
package top.bettercode.summer.security.token

import org.springframework.security.core.userdetails.UserDetails
import java.io.Serializable

/**
 * @author Peter Wu
 */
class ApiToken : Serializable {
    //--------------------------------------------
    lateinit var scope: String
    lateinit var accessToken: Token
    lateinit var refreshToken: Token
    lateinit var userDetailsInstantAt: InstantAt
    lateinit var userDetails: UserDetails

    constructor()
    constructor(
            scope: String, accessToken: Token,
            refreshToken: Token, userDetailsInstantAt: InstantAt, userDetails: UserDetails
    ) {
        this.scope = scope
        this.accessToken = accessToken
        this.refreshToken = refreshToken
        this.userDetailsInstantAt = userDetailsInstantAt
        this.userDetails = userDetails
    }

    val username: String
        get() = userDetails.username

    fun toApiToken(): ApiAccessToken {
        return ApiAccessToken(this)
    }

    companion object
}
